extensions/gdata/src/com/google/refine/extension/gdata/UploadCommand.java

```diff
@@ -147,7 +147,10 @@ private String upload(
 
     protected byte[] getIconImage() throws IOException {
         InputStream is = getClass().getResourceAsStream(METADATA_ICON_FILE);
-        return IOUtils.toByteArray(is);
+        byte[] iconImage = IOUtils.toByteArray(is);
+        is.close();
```

Comment:
@wetneb: you merged, but I think we have to release if an exception is raised by `IOUtils.toByteArray()`.

Your take?

Regards, Antoine

Reply:
@antoine2711 My thought is either way is OK since checked exceptions should be declared or thrown.

In both cases is.close() is guaranteed to be called. The finally is always executed regardless of whether an exception is thrown or caught

Reply:
This is extraordinarily unlikely to actually leak resources, but as long as it's being fixed, might as well fix it right.
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Reply:
Sure, having a finally block could be nice, sorry for the hasty merge!

Reply:
> Sure, having a finally block could be nice, sorry for the hasty merge!

@wetneb: what's the next step?
1. remodify this issue
2. create a new issue
3. forget about the next occasion?

Regards, Antoine

Reply:
Either 2. or 3.